packages/studio-be/src/studio/nlu/application/non-blocking-app.ts

```diff
@@ -0,0 +1,40 @@
+import ms from 'ms'
+import { NLUApplication } from '.'
+import { BotConfig } from './typings'
+
+const STAN_POLLING_INTERVAL = ms('1s')
+
+export class NonBlockingNluApplication extends NLUApplication {
+  private _waitingBots: BotConfig[] = []
+  private _stanIsReady = false
+
+  public async initialize() {
+    await this._waitForStan()
+    this._stanIsReady = true
+    this._logger.debug('Standalone NLU Server is ready.')
+    while (this._waitingBots.length) {
+      const bot = this._waitingBots.pop()!
+      await super.mountBot(bot)
+    }
+  }
+
+  public async mountBot(botConfig: BotConfig) {
+    if (this._stanIsReady) {
+      return super.mountBot(botConfig)
+    }
+    this._waitingBots.push(botConfig)
+  }
+
+  private _waitForStan() {
+    return new Promise(resolve => {
+      const i = setInterval(async () => {
+        const health = await this.getHealth({ reportError: false })
+        if (health?.isEnabled) {
+          clearInterval(i)
+          resolve()
+          return
+        }
+      }, STAN_POLLING_INTERVAL)
+    })
+  }
+}
```

Comment:
I don't think this logic is necessary. The studio should not wait after anything. When you press "Train now", it tries to contact the nlu server, if it's not up, just display an error message. We should stop making processes waiting for each other, it just adds complications

Reply:
The train now button can't be displayed If we don't know if a bot needs a training. To know if a bot needs a training, we have to query the nlu-server.

This is a long and complicated topic, but long story short: I submitted a new UI to the product team so that the studio UI would be a little more representative of the nlu-server API, but my suggestion was rejected. So yeah, we have to keep that.

Reply:
When you open the studio, you can make a query to the NLU server to check if training is required. I don't understand why the studio has to wait for the nlu server to be ready

Reply:
This is non-blocking for the studio, but anyway, I think you're right. 

nlu-server should be up when studio starts, if not we'll log an error.